webapp/src/app/shared/widgets/event-widget/event-editor/editor/event-editor.component.ts

```diff
@@ -103,8 +113,12 @@ export class EventEditorComponent implements OnChanges, OnInit {
 		this.detailsShown = false;
 	}
 	
-	export() {
-		return this.workingData?.map(event => event.export()) ?? this.eventData;
+	export(eventType?: EventArrayType, trader?: string): EventArray {
```

Comment:
Why is `EventArrayType` handled in the `EventEditorComponent` instead of the `NpcStatesComponent`? My understanding is that quest/trader etc is only relevant for NPC's so they should take care of properly saving it.